symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/AbstractDatafeedLoop.java

```diff
@@ -79,31 +84,37 @@ public void unsubscribe(RealTimeEventListener listener) {
    */
   protected void handleV4EventList(List<V4Event> events) {
     for (V4Event event : events) {
-      if (event == null || event.getType() == null) {
+
+      final Optional<RealTimeEventType> eventType = RealTimeEventType.fromV4Event(event);
+
+      if (!eventType.isPresent()) {
+        log.warn("Wrong V4Event received: {}", event);
         continue;
       }
 
-      try {
-        RealTimeEventType eventType = RealTimeEventType.valueOf(event.getType());
-        synchronized (listeners) {
-          for (RealTimeEventListener listener : listeners) {
-            if (listener.isAcceptingEvent(event, bdkConfig.getBot().getUsername())) {
-              eventType.dispatch(listener, event);
+      // dispatch single event using event's ID as traceId. Tested for DatafeedLoopV2 as well, and working.
+      DistributedTracingContext.doWithTraceId(event.getId(), () -> {
+
+        synchronized (this.listeners) {
+          for (RealTimeEventListener listener : this.listeners) {
+
+            if (listener.isAcceptingEvent(event, this.bdkConfig.getBot().getUsername())) {
+              try {
+                log.debug("Before dispatching '{}' event to listener {} (hashcode)", event.getType(), listener.hashCode());
```

Comment:
I thought that hashcode was going to be used for the trace id
If it is used to log something I would use the class simple name then

Reply:
Replaced by class name that is more useful. I'm still a bit sceptic of putting listener hashcode in MDC. Maybe too much?

Reply:
Yep, let's just use the even id for now